### PR TITLE
Contrib/hsue panw dlp integration1.2.0

### PR DESCRIPTION
**Pull request by henry-sue-pa**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Added support for authenticating with client credentials and support for sending notification using MS Teams

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: The parameter type for "access token" and "refresh token" is changed to "Authentication" so now they are treated as user name and password.
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml
Comment (ilappe) [inline Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml:7]: Because the new auth will use the Client ID & Client secret 
it's better to use those terms in the display
```suggestion
- display: Client ID
  displaypassword: Client Secret
```

**in addition** 
are the `client_id` and `client_secret` values the same as  the old values stored in `identifier` and `password`?
or user need to use another values ?

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP_test.py
Comment (ilappe) [inline Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP_test.py:218]: Why did you remove this?

File: Packs/Palo_Alto_Networks_Enterprise_DLP/ReleaseNotes/1_2_0.md
Comment (ilappe) [inline Packs/Palo_Alto_Networks_Enterprise_DLP/ReleaseNotes/1_2_0.md:6]: Due to the change in the credentials are breaking changes 
please create the release notes with the -bc flag 
`demisto-sdk update-release-notes .... -bc`
and please add a notes about the breaking changes
for example see [BC json file](Packs/AWS-GuardDuty/ReleaseNotes/1_2_6.json)

**Review by ilappe — CHANGES_REQUESTED** [review]
@henry-sue-pa thank you for contributing this content!!
please see the comments here

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP_test.py
Comment (henry-sue-pa) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP_test.py:218]: Because the test was failing due to framework level problem. If you know how to fix it, I can include it back.

Palo_Alto_Networks_Enterprise_DLP_test.py:349: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
Palo_Alto_Networks_Enterprise_DLP.py:588: in reset_last_run_command
    set_to_integration_context_with_retries(ctx)
CommonServerPython.py:8563: in set_to_integration_context_with_retries
    integration_context, version = update_integration_context(context, object_keys, sync)
CommonServerPython.py:8628: in update_integration_context
    latest_object = json.loads(integration_context.get(key, '[]'))
/usr/local/lib/python3.10/json/__init__.py:346: in loads
    return _default_decoder.decode(s)
/usr/local/lib/python3.10/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.decoder.JSONDecoder object at 0x7faf407de8c0>, s = '', idx = 0

    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.
    
        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.
    
        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

/usr/local/lib/python3.10/json/decoder.py:355: JSONDecodeError

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py
Comment (henry-sue-pa) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py:520]: Since we are pulling incidents every 5 seconds in this integration, our backend team is concerned about the performance impact it will have on our servers. They created an API for me to call every 5 minutes. This API will tell me how much time the integration should rest between each pull. If the server is healthy, it doesn't return anything. Otherwise it will return the # of seconds to rest. That is the reason for the if statement.

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml
Comment (henry-sue-pa) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml:7]: We are not replacing access token and refresh token with client id and client secrets. The support for client id and secret is in addition to access token/refresh tokens. So the user can choose to use either access token/refresh token or client id/client secret.

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Scripts/DlpAskFeedback/DlpAskFeedback.yml
Comment (henry-sue-pa) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Scripts/DlpAskFeedback/DlpAskFeedback.yml]: Updated comment.

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Scripts/DlpAskFeedback/DlpAskFeedback.py
Comment (henry-sue-pa) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Scripts/DlpAskFeedback/DlpAskFeedback.py]: Removed the comment

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py
Comment (henry-sue-pa) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py:229]: Updated the doc string

File: Packs/Palo_Alto_Networks_Enterprise_DLP/ReleaseNotes/1_2_0.md
Comment (henry-sue-pa) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/ReleaseNotes/1_2_0.md]: Updated release notes

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml
Comment (henry-sue-pa) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml:7]: https://user-images.githubusercontent.com/96203646/172691709-1db4d734-e30f-455c-b98b-71152f08f28c.mov

**Comment by ilappe** [comment]
@henry-sue-pa 
Thanks for your fixes !
did you saw @idovandijk's [comments](https://github.com/demisto/content/pull/19228#pullrequestreview-993824218)?

**Comment by idovandijk** [comment]
> @henry-sue-pa Thanks for your fixes ! did you saw @idovandijk's [comments](https://github.com/demisto/content/pull/19228#pullrequestreview-993824218)?

He saw them + we discussed them in the demo :)
@henry-sue-pa please lmk when done, I'll approve

**Comment by henry-sue-pa** [comment]
Hi Ido,

I am done with all the fixes. Please take a look and approve if you find no
more issues.

Henry

**Review by idovandijk — APPROVED** [review]
Once validations are satisfied feel free to merge

**Comment by henry-sue-pa** [comment]
@idovandijk 
The following validation errors can't be fixed because it is the only way I can find to support two modes of authentication. I have provided detail explanation and uploaded a screen recording for Israel. I will have to ask for an exemption on this.

[ERROR]: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml: [IN129] - You've removed integration parameters, the removed parameters are '{'refresh_token', 'access_token'}'

[ERROR]: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml: [IN116] - You've added required, the field is 'credentials'

**Comment by idovandijk** [comment]
> @idovandijk The following validation errors can't be fixed because it is the only way I can find to support two modes of authentication. I have provided detail explanation and uploaded a screen recording for Israel. I will have to ask for an exemption on this.
> 
> [ERROR]: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml: [IN129] - You've removed integration parameters, the removed parameters are '{'refresh_token', 'access_token'}'
> 
> [ERROR]: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml: [IN116] - You've added required, the field is 'credentials'

If you've added mandatory params to the integration then it breaks backward compatibility as far as I understand and requires a v2 integration... @ilappe this is your expertise :)

**Comment by ilappe** [comment]
@henry-sue-pa  as you added the json file in the release notes with the details about the BC change - it's enough 
i move it to doc review

**Review by ilappe — APPROVED** [review]

**Pull request merged**
